src/Workspaces/VisualBasic/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.vb

```diff
@@ -75,6 +75,30 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions.ContextQuery
         End Function
 
         <Extension()>
+        Public Function IsPartialTypeDeclarationNameContext(tree As SyntaxTree, position As Integer, cancellationToken As CancellationToken, ByRef statementSyntax As TypeStatementSyntax) As Boolean
+            If tree.IsInNonUserCode(position, cancellationToken) OrElse tree.IsInSkippedText(position, cancellationToken) Then
+                Return False
+            End If
+
+            Dim token = tree.FindTokenOnLeftOfPosition(position, cancellationToken) _
+                            .GetPreviousTokenIfTouchingWord(position)
+
+            Select Case token.Kind()
+                Case SyntaxKind.ClassKeyword,
+                     SyntaxKind.StructureKeyword,
+                     SyntaxKind.InterfaceKeyword,
+                     SyntaxKind.ModuleKeyword
+
+                    statementSyntax = token.GetAncestor(Of TypeStatementSyntax)()
+                    Return statementSyntax IsNot Nothing AndAlso
+                           statementSyntax.DeclarationKeyword = token AndAlso
+                           statementSyntax.Modifiers.Any(SyntaxKind.PartialKeyword)
```

Comment:
note: my comment about 'partial' only applies to C#.  For VB you do need this code.